### PR TITLE
Patch the new mechanoid weapons

### PR DESCRIPTION
**Pull request by SamaelGray**
## Additions

- Patched the new light weapons.
- Patched the Toxic Needle Gun.
- Tweaked the stats of the Hellsphere cannon and Graser.
- Patched the vanilla Needle Gun and Advanced Needle Gun from VFE-Mechs.
- Patched the mech-mounted Charge Blaster Turret.

## Additions

- Tweaked the sharp AP values of the 20 and 16 gauge shotgun shells.

## References

- https://docs.google.com/spreadsheets/d/1lbT0zzagCRPDJG4GctkeeoTsFCt3lYfM4SRnWt8ql-k/edit#gid=1573763037

## Reasoning

- The Slugthrower is similar to the vanilla assault rifle hence 5.56.
- The Spiner is a short very range gun, a new Spine projectile similar to 5.7x28mm without an ammoUser comp seemed fitting since the round is potent enough and  the War Urchin mechs don't stick around in the colony.
- Although the Mini-flameblaster works as intended even unpatched, it made sense to patch it similar to a normal CE flamethrower for better reliability.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long)

### CODE REVIEW

**Review by Tostov — CHANGES_REQUESTED** [review]
Regular needlegun is using caseless toxic ammoset instead of regular caseless. needs to be fixed.

**Review by N7Huntsman — DISMISSED** [review]
Some questions and feedback.

File: Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    <FireModes>
+      <aiAimMode>SuppressFire</aiAimMode>
+    </FireModes>
+    <weaponTags>
+      <li>CE_AI_AOE</li>
+      <li>NoSwitch</li>
+    </weaponTags>
+  </Operation>
+
+  <Operation Class="PatchOperationReplace">
```
[the remark this answers is not in the document]
Comment (SamaelGray) [reply Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml:188]: Cuz the operation fails to replace the custom verbClass of the flameblaster resulting in two attack verbs in-game...

File: Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      <SightsEfficiency>1</SightsEfficiency>
+      <ShotSpread>0.15</ShotSpread>
+      <SwayFactor>0.53</SwayFactor>
+      <Bulk>6.00</Bulk>
+    </statBases>
+    <Properties>
+      <recoilAmount>2.90</recoilAmount>
+      <verbClass>CombatExtended.Verb_ShootCE</verbClass>
+      <hasStandardCommand>true</hasStandardCommand>
+      <defaultProjectile>Bullet_16Gauge_Buck</defaultProjectile>
```
[the remark this answers is not in the document]
Comment (SamaelGray) [reply Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml:57]: As specifically mentioned in the reasoning section, 20 gauge is too weak and small for a _mini_ shotgun, it's not a micro shotgun and 12 gauge is too big for a mini shotgun that deals less damage than normal shotguns in vanilla and visibly fires less pellets.

File: Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      <warmupTime>0.8</warmupTime>
+      <range>16</range>
+      <burstShotCount>1</burstShotCount>
+      <soundCast>Shot_Shotgun_NoRack</soundCast>
+      <soundCastTail>GunTail_Heavy</soundCastTail>
+      <muzzleFlashScale>6</muzzleFlashScale>
+    </Properties>
+    <AmmoUser>
+      <magazineSize>10</magazineSize>
```
[the remark this answers is not in the document]
Comment (SamaelGray) [reply Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml:66]: Fair, I'd go for 6~8

File: Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+          <armorPenetrationBlunt>11</armorPenetrationBlunt>
+        </projectile>
+      </ThingDef>
+    </value>
+  </Operation>
+
+  <Operation Class="CombatExtended.PatchOperationMakeGunCECompatible">
+    <defName>Gun_Spiner</defName>
+    <statBases>
+      <RangedWeapon_Cooldown>0.37</RangedWeapon_Cooldown>
```
[the remark this answers is not in the document]
Comment (SamaelGray) [reply Biotech/Patches/ThingDefs_Misc/Weapons/RangedMechanoid_Light.xml:136]: The mech itself can't reload weapons regardless of having an ammoset or not, it's not smart enough on purpose. Can adjust the fire rate if it seems too high.

**Comment by Tostov** [comment]
Seems to be working without issues so far. Haven't run into the grazer yet to fully test, but based on the Diabolus I assume it's working mostly without issues.

**Review by N7Huntsman — APPROVED** [review]
Players might want a longer cooldown on the beamer depending upon how well the mech performs in combat, but given that the beam is only so-so currently due to how the vanilla verb works, I think it's fine for now.

**Review by N7Huntsman — DISMISSED** [review]
Fixed a typo.

**Review by albertbokor — APPROVED** [review]
Warqueen and entourage sufficiently fucks!

**Pull request merged**